ERCS/erc-7579.md

```diff
@@ -224,11 +234,35 @@ interface IModule {
     function onInstall(bytes calldata data) external;
     function onUninstall(bytes calldata data) external;
     function isModuleType(uint256 typeID) external view returns(bool);
+    function getModuleTypes() external view returns (uint256[]);
 }
 ```
 
 Modules MUST revert if `onInstall` or `onUninstall` was unsuccessful.
 
+
+##### Module State Management on Install and Uninstall**
+
+- When a module is **installed**, the smart account MUST initialize the module's state according to the module's specifications.
+- When a module is **uninstalled**, the smart account SHOULD clearly define whether the module's state is preserved or reset. This behavior MUST be documented in the module's interface.
+
+This ensures that upon reinstallation of a module, its state does not inadvertently carry over from its previous installation unless explicitly designed to do so.
+
+#### Mandatory Presence of a Validation Module**
```

Comment:
duplicate of above

Reply:
It's not the same thing. One concerns state reset while the other relates to the presence of a Validation module. I'll improve it for more clarity.

Reply:
I meant the other section **Mandatory Presence of a Validation Module** above